theme/static/js/hs-vue/relevant-tools.js

```diff
@@ -0,0 +1,136 @@
+                    ';tool_res_id=' + tool.res_id + ';res_id=' + this.resId;
+            }
+            return null;    // default
+        }
+    },
+    mounted: function () {
+        let vue = this;
+        $.get("/hsapi/_internal/" + this.resId + "/relevant-tools/", function (response) {
+            response = JSON.parse(response);
+            if (response) {
+                vue.tools = response['tool_list'].map(function (tool) {
+                    // We need to encode special characters in Urls
+                    if (tool.hasOwnProperty('url')) {
+                        tool.url = encodeURIComponent(tool.url);
+                    }
+                    else if (tool.hasOwnProperty('url_aggregation')) {
+                        tool.url_aggregation = encodeURIComponent(tool.url_aggregation);
+                    }
+                    else if (tool.hasOwnProperty('url_file')) {
+                        tool.url_file = encodeURIComponent(tool.url_file);
+                    }
+
+                    return tool;
+                });
+
+                vue.openWithTools = vue.tools.filter(function(tool) {
+                    return tool.url && tool.openwithlist;
+                });
+
+                // Append menu items to right click menu in file browser
+                let menu = $("#right-click-menu");
+                let hasTools = false;
+                for (let i = 0; i < vue.tools.length; i++) {
+                    let tool = vue.tools[i];
+
+                    if (!tool.openwithlist || !tool.url) {
+                        continue;   // Ignore tools not listed in OpenWith menu
+                    }
+
+                    if (tool['agg_types']) {
+                        let aggregationUrl = vue.getAggregationAppUrl(tool);
+                        if (aggregationUrl) {
+                            let menuItem =
+                            '<li class="btn-open-with" data-menu-name="web-app" ' +
+                                'data-agg-types="' + tool['agg_types'] + '" data-url-aggregation="' +
+                                aggregationUrl + '">' +
+                                '<img class="file-options-webapp-icon" src="' + tool['icon_url'] +
+                                    '" alt="' + tool['title'] + '"/>' + '<span>' + tool['title'] + '</span>' +
+                            '</li>';
+                            if (!hasTools) {
+                                menu.append('<li id="tools-separator" role="separator" class="divider"></li>');
+                            }
+                            menu.append(menuItem);
+                            hasTools = true;
+                        }
+                    }
+
+                    if (tool['file_extensions']){
+                        let urlFile = vue.getFileAppUrl(tool);
+                        if (urlFile) {
+                            let menuItem =
+                            '<li class="btn-open-with" data-menu-name="web-app" ' +
```

Comment:
Not for now, since we haven't instantiated the File Browser for Vue. But I plan to do that in the near future. I can't think of a better method for now since these data are now loaded asynchronously and so we can't use Django template language to render these.